website/blog/2022-05-10-acf-in-a-project.md

```diff
@@ -0,0 +1,140 @@
+
+```bash
+wp boilerplate create_acf_meta --name=intro
+```
+
+This command will generate a **_CustomMeta_** folder inside **_src_** folder and add a new file called **_IntroAcfMeta.php_**. Inside that file, you should see the following method:
+```php
+public function fields(): void
+{
+	if (function_exists('acf_add_local_field_group')) {
+		\acf_add_local_field_group([]);
+	}
+}
+```
+
+Go back to your PHP code export for ACF fields and select everything inside the `acf_add_local_field_group()` function and paste it into your function. To make it in line with our coding standards, we have to do the following:
+- replace `array()` with `[]`
+- make every label translatable and escaped - use `esc_html__()`
+- instead of hardcoding the field name, replace it with a class constant
+
+After all these changes, your code should look like this:
+
+```php
+class IntroAcfMeta extends AbstractAcfMeta
+{
+
+	/**
+	 * Intro field name.
+	 */
+	public const INTRO_FIELD_NAME = 'intro';
+	
+	/**
+	 * Render acf fields.
+	 *
+	 * @return void
+	 */
+	public function fields(): void
+	{
+		if (function_exists('acf_add_local_field_group')) {
+			\acf_add_local_field_group([
+				'key' => 'group_6269300acefda',
+				'title' =>  \esc_html__('Intro', 'eightshift-theme'),
+				'fields' => [
+					[
+						'key' => 'field_6269300f8029b',
+						'label' => \esc_html__('Intro', 'eightshift-theme'),
+						'name' => self::INTRO_FIELD_NAME,
+						'type' => 'text',
+						// ...
+					]
+				]
+				// ...
+			]);
+		}
+	}
+}
+```
+
+The final step is to go back to Custom Fields in WP Admin and delete your field group from there. Your field should now be visible when editing Posts.
+
+## Using get_field()
+
+To fetch the saved meta value, we use ACF's `get_field()` function, but here are a few tips that could improve your code quality. First, you should check if that function exists. If ACF gets for whatever reason deactivated on your site, it won't break your site. The second one is using class constant instead of hardcoding the field name. With these practices in mind, your code should look like this:
+
+```php
+use YourNamespace\CustomMeta\IntroAcfMeta;
+
+// ...
+
+if (function_exists('get_field')) {
+	$introText = get_field(IntroAcfMeta::INTRO_FIELD_NAME, $postId);
+}
+```
+
+> It's better to use class constants because if you decide to change the field name, you will have to change it only in one place.
+
+## Theme Options
+
+ACF's Options page has a wide array of uses and there's a very likely chance you may also need some sort of Theme Options in your project. To make the implementation of Theme Options a bit easier, we have a CLI command which generates the `ThemeOptions` class in your project. Just use the following command:
```

Comment:
```suggestion
ACF's Options page has a wide array of uses and it's very likely that you'll need some sort of Theme Options in your project. To make the implementation of Theme Options a bit easier, we have a CLI command which generates the `ThemeOptions` class in your project. Just use the following command:
```

I thought we should use `[]` instead of `array` though 😂

Reply:
It took me a while to figure out what you mean by that 😅  nice one! 😄